src/community/mbstyle/src/main/java/org/geoserver/community/mbstyle/web/MBStyleHandler.java

```diff
@@ -0,0 +1,112 @@
+/* (c) 2017 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.community.mbstyle.web;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.util.List;
+
+import org.apache.commons.io.FilenameUtils;
+import org.geoserver.catalog.SLDHandler;
+import org.geoserver.catalog.StyleHandler;
+import org.geoserver.platform.resource.FileSystemResourceStore;
+import org.geoserver.platform.resource.Resource;
+import org.geotools.mbstyle.MapBoxStyle;
+import org.geotools.styling.ResourceLocator;
+import org.geotools.styling.StyledLayerDescriptor;
+import org.geotools.util.Version;
+import org.json.simple.parser.ParseException;
+import org.xml.sax.EntityResolver;
+
+public class MBStyleHandler extends StyleHandler {
+
+    public static final String FORMAT = "json";
+
+    public static final String MIME_TYPE = "application/vnd.geoserver.json";
+
+    private SLDHandler sldHandler;
+
+    protected MBStyleHandler(SLDHandler sldHandler) {
+        super("JSON", FORMAT);
+        this.sldHandler = sldHandler;
```

Comment:
Name is used by the GeoServer UI to display the style, so should be something like MBStyle rather than "JSON"